core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -132,12 +134,40 @@ private CarbonTable() {
     this.tablePrimitiveDimensionsMap = new HashMap<String, List<CarbonDimension>>();
   }
 
+  /**
+   * During creation of TableInfo from hivemetastore the DataMapSchemas and the columns
+   * DataTypes are not converted to the appropriate child classes.
+   *
+   * This method will cast the same to the appropriate classes
+   *
+   * @param tableInfo
+   */
+  private static void updateTableInfo(TableInfo tableInfo) {
+    if (Boolean.parseBoolean(CarbonProperties.getInstance().getProperty(
+        CarbonCommonConstants.ENABLE_HIVE_SCHEMA_META_STORE))) {
```

Comment:
Irrespective enabling hivemetastore better always do this conversion here only. Remove this conversion from other places